app/views/events/index.html.erb

```diff
@@ -0,0 +1,23 @@
+<h2>All Events</h2>
+
+<p>All Upcoming Events</p>
+<% Event.upcoming.each do |event| %>
```

Comment:
Please retrieve this collection into a variable in your controller and only use the variable  here